backend/dataall/modules/shares_base/services/sharing_service.py

```
@@ -295,64 +358,98 @@ def reapply_share(cls, engine: Engine, share_uri: str) -> bool:
-                                    session, share_data, shareable_items
-                                ).process_approved_shares()
-                                log.info(f'Reapplying {type.value} succeeded = {success}')
-                                if not success:
-                                    reapply_successful = False
-                            else:
-                                log.info(f'There are no items to reapply of type {type.value}')
-                        except Exception as e:
-                            log.error(f'Error occurred during share reapplying of {type.value}: {e}')
-
-                return reapply_successful
-
-            except ResourceLockTimeout as e:
-                ShareStatusRepository.update_share_item_health_status_batch(
-                    session,
-                    share_uri,
-                    old_status=ShareItemHealthStatus.PendingReApply.value,
-                    new_status=ShareItemHealthStatus.Unhealthy.value,
-                    message=str(e),
-                )
+                            ShareNotificationService(
+                                session=session, dataset=share_data.dataset, share=share_data.share
+                            ).notify_share_object_items_healthy()
 
-            except Exception as e:
-                log.exception('Error occurred during share approval')
-                return False
+                except ResourceLockTimeout as timeout_exception:
+                    ShareStatusRepository.update_share_item_health_status_batch(
+                        session,
+                        share_uri,
+                        old_status=ShareItemHealthStatus.PendingReApply.value,
+                        new_status=ShareItemHealthStatus.Unhealthy.value,
+                        message=str(timeout_exception),
+                    )
+                    code_exception_list.append(str(timeout_exception))
+        except Exception as e:
```

should you raise the exception?

---

Not raising an exception and handling it here in the service layer itself. The caller of this method will get to know if this failed or not failed based on if this function returns True/False.